app/controllers/exercises_controller.rb

```diff
@@ -53,4 +53,23 @@ def show
       end
     end
   end
+
+  def set_disabled_statuses
+    @course = Course.find(params[:course_id])
+    @organization = @course.organization
+    authorize! :teach, @organization
+
+    action = params[:commit] == 'Disable selected' ? :disabled! : :enabled!
+    exercise_params = params[:exercise]
+    exercise_names = []
+    exercise_names = exercise_params.map { |k, v| v == '1' ? k : nil }.compact unless exercise_params.nil?
+
+    exercise_names.each do |name|
+      exercise = Exercise.find_by(name: name)
+      exercise.send(action) unless exercise.nil?
+    end
```

Comment:
There must be a more effective way for this. Here we are creating N queries to DB.